mlir/include/mlir/Dialect/MPI/IR/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+add_mlir_dialect(MPI mpi)
+add_mlir_doc(MPI MPI Dialects/ -gen-dialect-doc)
+
+# Add MPI operations
+set(LLVM_TARGET_DEFINITIONS MPIOps.td)
+mlir_tablegen(MPIOps.h.inc -gen-op-decls)
+mlir_tablegen(MPIOps.cpp.inc -gen-op-defs)
+add_public_tablegen_target(MLIRMPIOpsIncGen)
+add_dependencies(mlir-generic-headers MLIRMPIOpsIncGen)
```

Comment:
mlir-generic-headers are dialect independent. I don't think we should be adding a dependency of MPI dialect on them.

Reply:
I don't think this was resolved

Reply:
Good catch! Cleaned the last one up too!